lib/internal/http2/util.js

```diff
@@ -350,6 +354,67 @@ function getSettings(session, remote) {
 
 function updateSettingsBuffer(settings) {
   let flags = 0;
+  let numCustomSettings = 0;
+
+  if (typeof settings.customSettings === 'object') {
+    const customSettings = settings.customSettings;
+    for (const setting in customSettings) {
+      const val = customSettings[setting];
+      if (typeof val === 'number') {
+        let set = false;
+        const nsetting = Number(setting);
+        if (nsetting < IDX_SETTINGS_FLAGS) {
+          set = true;
+          switch (nsetting) {
```

Comment:
Is the switch statement really necessary? This could be potentially be simplified to just:

```js
flags |= (1 << nsetting);
settingsBuffer[nsetting] = value;
```

Without breaking things out with the switch statement.

Reply:
I have to check, whether the setting is one of the nghttp2 supported settings.
Alternative, I could also check, if the setting is in an const array of supported settings, but I assumed, that in this way the jit compiler may generate faster code and that one might add special handling for some settings, if it will be necessary.

Reply:
There is another issue. I have read several specs, and might be that not all supported SETTINGS ids will be a continuous range of integers in the future. Otherwise, I could also simply check, if it is within the supported ids.

Reply:
ok, that makes sense to keep it like it is then